core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -68,6 +71,21 @@ private[spark] object TaskDescription {
     }
   }
 
+  private def serializeResources(map: immutable.Map[String, ResourceInformation],
+      dataOut: DataOutputStream): Unit = {
+    dataOut.writeInt(map.size)
```

Comment:
Should the ResourceInformation object just be serializable and manage this directly?

Reply:
I checked the commit history and seems we intend to hand write the SerDe code for TaskDescription class, and in this way we only able to serialize simple data types(Int、UTF etc.) but not able to serialize an object directly.

Reply:
Is it for Python support?